src/integrationTest/java/uk/gov/hmcts/reform/cwrdapi/UpdateStaffReferenceProfileTest.java

```diff
@@ -403,6 +407,62 @@ void should_return_update_staff_user_with_status_code_400_invalid_roles() throws
 
     }
 
+    @Test
+    void should_return_reinvite_staff_user_with_status_code_404_profile_doesnot_exist() throws Exception {
+
+        StaffProfileCreationRequest request = caseWorkerReferenceDataClient.createStaffProfileCreationRequest();
+        request.setEmailId("test@test.com");
+        request.setResendInvite(true);
+
+        Map<String, Object> response = caseworkerReferenceDataClient
+                .updateStaffProfile(request,ROLE_STAFF_ADMIN);
+
+        assertThat(response.get("http_status")).isEqualTo("404");
+        String responseBody = (String) response.get("response_body");
+        assertThat(responseBody.contains(INVALID_PROFILE)).isTrue();
+
+    }
+
+    @Test
+    void should_return_reinvite_staff_user_with_status_code_200_profile() throws Exception {
+
+        StaffProfileCreationRequest request = caseWorkerReferenceDataClient.createStaffProfileCreationRequest();
+        userProfilePostUserWireMockForStaffProfile(HttpStatus.CREATED);
+        request.setResendInvite(true);
+
+        Map<String, Object> createResponse = caseworkerReferenceDataClient.createStaffProfile(request,ROLE_STAFF_ADMIN);
+        Map createBody = (Map)createResponse.get("body");
+        Map<String, Object> resendResponse = caseworkerReferenceDataClient.updateStaffProfile(request,ROLE_STAFF_ADMIN);
+
+        assertThat(resendResponse).isNotNull();
+        assertThat(resendResponse.get("http_status")).isEqualTo("200 OK");
+        Map resendResponseBody = (Map) resendResponse.get("body");
+        assertEquals(createBody.get("case_worker_id"), resendResponseBody.get("case_worker_id"));
+    }
+
+    @Test
+    void should_update_IdamId_when_reinvite_staff_user_true_in_crd() throws Exception {
+
+        StaffProfileCreationRequest request = caseWorkerReferenceDataClient.createStaffProfileCreationRequest();
+        userProfilePostUserWireMockForStaffProfile(HttpStatus.CREATED);
+        request.setResendInvite(true);
+
+        Map<String, Object> createResponse = caseworkerReferenceDataClient.createStaffProfile(request,ROLE_STAFF_ADMIN);
+        Map createBody = (Map)createResponse.get("body");
+        Map<String, Object> resendResponse = caseworkerReferenceDataClient.updateStaffProfile(request,ROLE_STAFF_ADMIN);
+
+        assertThat(resendResponse).isNotNull();
+        assertThat(resendResponse.get("http_status")).isEqualTo("200 OK");
+        Map resendResponseBody = (Map) resendResponse.get("body");
+        assertEquals(createBody.get("case_worker_id"), resendResponseBody.get("case_worker_id"));
+
+        String path = "/profile/search-by-name";
+        ResponseEntity<SearchStaffUserResponse[]> fetchstaff = caseworkerReferenceDataClient
+                .searchStaffUserByNameExchange(path, request.getFirstName(), "1", "1",
+                        ROLE_STAFF_ADMIN);
+        assertEquals(resendResponseBody.get("case_worker_id"), fetchstaff.getBody()[0].getCaseWorkerId());
```

Comment:
can you add another assert to verify createResponse case_worker_id is not same as fetchstaff case_worker_id

Reply:
done Thnak you @rajkatla-hmcts